core/src/main/scala/org/apache/spark/rpc/netty/Inbox.scala

```diff
@@ -200,6 +200,17 @@ private[netty] class Inbox(val endpointName: String, val endpoint: RpcEndpoint)
    * Calls action closure, and calls the endpoint's onError function in the case of exceptions.
    */
   private def safelyCall(endpoint: RpcEndpoint)(action: => Unit): Unit = {
+    def dealWithFatalError(fatal: Throwable): Unit = {
+      inbox.synchronized {
+        // Should reduce the number of active threads before throw the error
+        if (numActiveThreads > 0) {
```

Comment:
if the condition is not going to happen, I'd use `assert(numActiveThreads > 0)` instead of `if`

Reply:
Thanks for the advice, updated.